server/app.js

```diff
@@ -1,15 +1,260 @@
+        });
+      }
+    })
+    .catch((error) => {
+      res.status(500);
+      res.json({
+        error: `Internal Server Error: ${error}`,
+      });
+    });
+});
+
+// Get all students in one Course
+app.get("/courses/:courseId/students", (req, res) => {
+  const { courseId } = req.params;
+  Student.find({ course: courseId })
+    .then((students) => {
+      if (student.course === courseId) {
+        res.jsonp(students);
+        res.status(200);
+      } else {
+        res.status(404);
+        res.json({
+          error: `Post with id: ${courseId} not found`,
+        });
+      }
+    })
+    .catch((error) => {
+      res.status(500);
+      res.json({
+        error: `Internal Server Error: ${error}`,
+      });
+    });
+});
+
+// most specific: find one student in one course
+app.get("/courses/:courseId/students/:studentId", (req, res) => {
+  const { courseId, studentId } = req.params;
+  Student.findById(courseId && studentId)
+    .then((student) => {
+      if (student.course === courseId) {
+        res.status(200);
+        res.json(student);
+      } else {
+        res.status(404);
+        res.json({
+          error: `Student with id: ${studentId} not found`,
+        });
+      }
+    })
+    .catch((error) => {
+      res.status(500);
+      res.json({
+        error: `Internal Server Error: ${error}`,
+      });
+    });
+});
+
+// find all students
+app.get("/students", (req, res) => {
+  Student.find()
+    .then((students) => {
+      res.status(200);
+      res.json(students);
+    })
+    .catch((error) => {
+      res.status(500);
+      res.json({
+        error: `Internal Server Error: ${error}`,
+      });
+    });
+});
+
+// find specific student
+app.get("/students/:studentId", (req, res) => {
+  const { studentId } = req.params;
+  Student.findById(studentId)
+    .then((student) => {
+      if (student) {
+        res.status(200);
+        res.json(student);
+      } else {
+        res.status(404);
+        res.json({
+          error: `Post with id: ${id} not found`,
+        });
+      }
+    })
+    .catch((error) => {
+      res.status(500);
+      res.json({
+        error: `Internal Server Error: ${error}`,
+      });
+    });
+});
+
+//create new courses
+app.post("/courses", (req, res) => {
+  Course.create(req.body)
+    .then((newCourse) => {
+      if (newCourse.location === undefined) {
+        newCourse.location = "Remote";
+      }
+      res.status(201);
+      res.json(newCourse);
+    })
+
+    .catch((error) => {
+      console.error(error);
+    });
+});
+
+// create new students
+app.post("/courses/:courseId/students", (req, res) => {
+  const { courseId } = req.params;
+  Student.create({
+    firstName: req.body.firstName,
+    lastName: req.body.lastName,
+    course: courseId,
+  })
+    .then((newStudent) => {
+      res.status(201);
+      res.json(newStudent);
+    })
+
+    .catch((error) => {
+      console.error(error);
```

Comment:
We should send a response whenever an errror happens, otherwise this request will never receive an response